### PR TITLE
[Structural] minor beam cleanup

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
@KlausBSautter I noticed several leftovers in the Beam. Please check it closely, this is a wip

These changes are also necessary for the new beam from what I saw

### CODE REVIEW

**Comment by loumalouomega** [comment]
FYI @Joekchu

File: applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp [hunk 1]
```diff
@@ -885,8 +882,7 @@ void TrussElement3D2N::CalculateElasticStiffnessMatrix(
 void TrussElement3D2N::FinalizeSolutionStep(const ProcessInfo& rCurrentProcessInfo)
 {
     KRATOS_TRY;
-    ProcessInfo temp_process_information;
-    ConstitutiveLaw::Parameters Values(GetGeometry(),GetProperties(),temp_process_information);
+    ConstitutiveLaw::Parameters Values(GetGeometry(),GetProperties(),rCurrentProcessInfo);
```
Comment (KlausBSautter) [inline applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp:899]: does this work?
I remember that we had to create a non-const ProcessInfo to pass

Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp:899]: possible since #8090

**Comment by Joekchu** [comment]
> FYI @Joekchu

Thanks a lot!  :)

File: applications/StructuralMechanicsApplication/custom_elements/cable_element_3D2N.cpp [hunk 2]
```diff
@@ -120,8 +120,8 @@ void CableElement3D2N::UpdateInternalForces(
     }
 
     Vector temp_internal_stresses = ZeroVector(msLocalSize);
-    ProcessInfo temp_process_information;
-    ConstitutiveLaw::Parameters Values(GetGeometry(),GetProperties(),temp_process_information);
+    // ProcessInfo temp_process_information; this mus be passed to the function!
```
Comment (rubenzorrilla) [inline applications/StructuralMechanicsApplication/custom_elements/cable_element_3D2N.cpp:123]:
```suggestion
    // ProcessInfo temp_process_information; this must be passed to the function!

```

**Comment by rubenzorrilla** [comment]
BTW, what about the new beam @KlausBSautter ? Did you find any time to work on the 2D / validation?

File: applications/StructuralMechanicsApplication/custom_elements/cable_element_3D2N.cpp (hunk 2 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_elements/cable_element_3D2N.cpp:123]: this is a WIP comment for @KlausBSautter, it should be removed before merging

**Comment by KlausBSautter** [comment]
@rubenzorrilla Philipp tested in 3d but ran into some problems, we'll look into that. But as far as the 2D beam is concerned I am waiting for you to tests it.

**Comment by KlausBSautter** [comment]
@philbucher this should compile now. For some reasons only 49/980 cpp tests run, though

**Comment by KlausBSautter** [comment]
Initializing KratosStructuralMechanicsApplication...
Importing    KratosConstitutiveLawsApplication 
Initializing KratosConstitutiveLawsApplication...
.................................................
Ran 49 of 980 test cases in 0.00867541s. OK

**Comment by philbucher** [comment]
> @philbucher this should compile now. For some reasons only 49/980 cpp tests run, though

those are only the ones of StructuralMechanics

**Review by KlausBSautter — APPROVED** [review]

**Comment by philbucher** [comment]
thx @KlausBSautter

**Pull request merged**